agent/osv_agent.py

```diff
@@ -118,16 +119,51 @@ def process(self, message: m.Message) -> None:
         Once the scan is completed, it emits messages of type : `v3.report.vulnerability`
         """
         logger.info("processing message of selector : %s", message.selector)
-        content = _get_content(message)
-        if content is None or content == b"":
-            logger.warning("Message file content is empty.")
-            return
-        for file_name in SUPPORTED_OSV_FILE_NAMES:
-            scan_results = _run_osv(file_name, content)
-            if scan_results is not None:
-                logger.info("Found valid name for file: %s", file_name)
-                self._emit_results(scan_results)
-                break
+        if message.selector == "v3.asset.file":
+            content = _get_content(message)
+            if content is None or content == b"":
+                logger.warning("Message file content is empty.")
+                return
+            for file_name in SUPPORTED_OSV_FILE_NAMES:
+                scan_results = _run_osv(file_name, content)
+                if scan_results is not None:
+                    logger.info("Found valid name for file: %s", file_name)
+                    self._emit_results(scan_results)
+                    break
+
+        elif message.selector in [
+            "v3.fingerprint.file.android.library",
+            "v3.fingerprint.file.ios.library",
+            "v3.fingerprint.file.library",
+        ]:
+            package_name = message.data.get("library_name")
+            package_version = message.data.get("library_version")
+            package_type = message.data.get("library_type")
+
+            api_result = osv_service_api.query_osv_api(
+                package_name=package_name,
+                version=package_version,
+                ecosystem=package_type,
```

Comment:
The `library_type` sent by the other agents is not necessarily the ecosystem the osv API is execpecting.